estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,114 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from datetime import date, timedelta
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+    _order = "price desc"
+
+    price = fields.Float(string="Price")
+    status = fields.Selection(
+        selection=[("accepted", "Accepted"), ("refused", "Refused")],
+        string="Status",
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one(
+        "estate.property",
+        string="Property",
+        required=True,
+        ondelete="cascade",
+    )
+    property_type_id = fields.Many2one(
+        comodel_name="estate.property.type",
+        string="Property Type",
+        related="property_id.property_type_id",
+        store=True,
+    )
+
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline", store=False)
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self) -> None:
+        for offer in self:
+            created = (offer.create_date or fields.Datetime.now())
+            base_date = created.date() if hasattr(created, "date") else date.today()
+            offer.date_deadline = base_date + timedelta(days=offer.validity or 0)
+
+    def _inverse_date_deadline(self) -> None:
+        for offer in self:
+            created = (offer.create_date or fields.Datetime.now())
+            base_date = created.date() if hasattr(created, "date") else date.today()
```

Comment:
```suggestion
            created = (offer.create_date or fields.Datetime.now())
            base_date = created.date()
```
I think that with your first line you would not have an error becasue created should be a datetime in any case ?